src/modules/logger/log_writer_file.cpp

```diff
@@ -218,20 +199,37 @@ void LogWriterFile::start_log(LogType type, const char *filename)
 	}
 
 #if PX4_CRYPTO
-	bool enc_init = init_logfile_encryption(filename);
+	uint8_t *keydata;
+	size_t keydatasize;
+	bool enc_init = init_logfile_encryption(&keydata, &keydatasize);
 
 	if (!enc_init) {
 		PX4_ERR("Failed to start encrypted logging");
 		_crypto.close();
+
+		if (keydata) {
+			free(keydata);
+		}
+
 		return;
 	}
 
-#endif
```

Comment:
I was just wondering; would it simply work if you pushed the keydata into buffer right after start_log here? That is,
	_buffers[(int)type].write_no_check(keydata, keydatasize);

?

Then you didn't need to pass the keydata down to the start_log, or modify that?

Just an idea, I don't know if it would work or not ;)

Reply:
Ah.. sorry, now I understood what did you mean. That would be much much better! I'll check if that works.